### PR TITLE
【Hackathon4】No.5 为 Paddle 新增 nextafter API

### PR DESCRIPTION
**Pull request by enkilee**
为 Paddle 新增 nextafter API

### CODE REVIEW

File: rfcs/APIs/20230327_api_design_for_nextafter.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+该 API 实现于 `paddle/phi/kernels/funcs/elementwise_functor.h`，
+
+通过调研发现，需要
+1 `python/paddle/tensor/math.py` 添加Python 实现代码 & 英文 API 文档，
+2 `python/paddle/tensor/init.py` 添加 nextafter API，以支持 Tensor.nextafter 的调用方式。
+3 `paddle/phi/kernels/elementwise_kernel.cc` 添加底层OP。
+4 `paddle/phi/kernels/` 目录下添加`elementwise_nextafter_kernel.h`、`elementwise_nextafter_grad_kernel.h`文件。
+5 `paddle/phi/kernels/cpu` 目录下添加`elementwise_nextafter_kernel.cc`、`elementwise_nextafter_grad_kernel.cc`文件。
+6 `paddle/phi/kernels/funcs/elementwise_functor.h` 添加C++实现代码。
```
Comment (jeff41404) [inline rfcs/APIs/20230327_api_design_for_nextafter.md:96]: 涉及到C++算子开发，可以参考官网文档[开发 C++ 算子](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/dev_guides/api_contributing_guides/new_cpp_op_cn.html)，这里至少要修改算子描述及定义等yaml配置文件，同时命名可以简单点，就叫 nextafter_kernel.h/cc 即可

File: rfcs/APIs/20230327_api_design_for_nextafter.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+# 六、测试和验收的考量
+
+测试需要考虑的 case 如下：
+
+- 输出数值结果的一致性和数据类型是否正确，使用 pytorch 作为参考标准
```
Comment (jeff41404) [inline rfcs/APIs/20230327_api_design_for_nextafter.md:102]: can only be compared with numpy in unittest

**Comment by enkilee** [comment]
@jeff41404 已修改

File: rfcs/APIs/20230327_api_design_for_nextafter.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+## PyTorch
+
+PyTorch 中有 `torch.nextafter` 的API，详细参数为 `torch.nextafter(input, other, *, out=None) → Tensor`。
+
+在 PyTorch 中的介绍为：
+
+>Return the next floating-point value after input towards other, elementwise.The shapes of input and other must be broadcastable.
+
+在实现方法上，PyTorch 是通过 C++ API 组合实现的，[代码位置](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/cpu/vec/vec_base.h#L473-L479)
```
Comment (jeff41404) [inline rfcs/APIs/20230327_api_design_for_nextafter.md:39]: it is better to describe that `torch.nextafter` is implemented in terms of  `std:: nextafter`?

File: rfcs/APIs/20230327_api_design_for_nextafter.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+该 API 实现于 `paddle/phi/kernels/funcs/elementwise_functor.h`，
+
+通过调研发现，需要
+1 `paddle/phi/api/yaml/ops.yaml` 添加前向算子，`paddle/phi/api/yaml/backward.yaml` 添加反向算子。
+2 `paddle/phi/infermeta/generated.cc` 添加算子 InferMeta。
+3 `paddle/phi/kernels/elementwise_kernel.cc` 添加底层OP。
+4 `paddle/phi/kernels/` 目录下添加`nextafter_kernel.h`、`nextafter_grad_kernel.h`文件。
+5 `paddle/phi/kernels/cpu` 目录下添加`nextafter_kernel.cc`、`nextafter_grad_kernel.cc`文件。
+6 `paddle/phi/kernels/funcs/elementwise_functor.h` 添加C++实现代码。
```
Comment (jeff41404) [inline rfcs/APIs/20230327_api_design_for_nextafter.md:96]: There are no GPU related implementation files here, really not needed？

File: rfcs/APIs/20230327_api_design_for_nextafter.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+7 `python/paddle/tensor/math.py` 添加Python 实现代码 & 英文 API 文档，
+8 `python/paddle/tensor/init.py` 添加 nextafter API，以支持 Tensor.nextafter 的调用方式。
+9 `python/paddle/fluid/tests/unittests` 目录下添加单测文件test_nextafter_op.py。
```
Comment (jeff41404) [inline rfcs/APIs/20230327_api_design_for_nextafter.md:99]: the formatting of markdown of these 9 lines is not right?  all displayed on the same line.

**Comment by enkilee** [comment]
@jeff41404  MD file already changed.

**Comment by jeff41404** [comment]
Does this API have backward logic? also need to explanation

**Comment by enkilee** [comment]
@jeff41404 done

**Review by jeff41404 — APPROVED** [review]
LGTM

**Pull request merged**